### ISSUE TITLE
Downlevel TemplateStringsArray for tagged templates should be frozen

### ISSUE THREAD

**Issue by rbuckton** [microsoft/TypeScript#51789]
# Bug Report

The `TemplateStringsArray` we emit for tagged templates (as well as its `raw` property) should be frozen when we emit to `--target ES5`, to match the ES2015 specification. Without this change, developers could run into issues with a spec-compliant [`String.dedent`](https://github.com/tc39/proposal-string-dedent) polyfill. This was first reported here: https://github.com/tc39/proposal-string-dedent/issues/75#issuecomment-1333011358

### 🔎 Search Terms

tagged template array frozen TemplateStringsArray

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?target=1#code/CYUwxgNghgTiAEA3W8BmAueUB2BPA3AFCGoAGUAJAN4CMAvgEalA)

### 💻 Code

```ts
const f = (ar: TemplateStringsArray, ...args) => ar;
const x = f`a${1}b`
console.log(Object.isFrozen(x));
console.log(Object.isFrozen(x.raw));
```

### 🙁 Actual behavior

Neither the `TemplateStringsArray` nor its `raw` property are frozen.

### 🙂 Expected behavior

Both the `TemplateStringsArray` and its `raw` property should be frozen.

**Comment by fatcerberus** [comment]
Unrelated but I really don’t like that the proposal wants to strip blank lines completely.  That feels like more than what a function called `dedent` should be doing.

**Comment by bakkot** [comment]
@fatcerberus You should open an issue on that repo there if you want to discuss it; this thread isn't the right place. (Though I think you may have misread the proposal.)

**Comment by fatcerberus** [comment]
Yes, I know this isn't the right place to discuss it.  That's why I said "unrelated".  It was more idle musing than anything; I don't actually have a strong enough opinion one way or the other to bother formally arguing the point.

tbh I'm more curious exactly what this function is doing under the hood that it apparently *requires* a frozen TemplateStringsArray.

**Comment by DanielRosenwasser** [comment]
We won't be addressing this - lowest supported ES target for TypeScript 6 is ES2015.
